### ISSUE TITLE
ElasticSearch 6 Doc Manager?

### ISSUE THREAD

**Issue by aguyinmontreal** [mongodb-labs/mongo-connector#770]
Can we install the elastic5 doc manager with ElasticSearch 6 ?

https://github.com/mongodb-labs/mongo-connector/wiki/Usage%20with%20ElasticSearch

**Comment by cncap** [comment]
I also want to know that. thanks

**Comment by aguyinmontreal** [comment]
@cncap   as I said on issue #758 , I don't think so, because [ElasticSearch 6 doesn't allow "types" anymore](https://www.elastic.co/guide/en/elasticsearch/reference/6.x/removal-of-types.html), so I really don't know how mongo-connector will handle "collections" now!

**Comment by escaroda** [comment]
Yeah. ElasticSearch have changed data structure now. Mongo-connector will work sooner or later but it requires to write a new Doc Manager for ElasticSearch 6 (and look forward to ElasticSearch 8 according to this roadmap [https://www.elastic.co/guide/en/elasticsearch/reference/6.x/removal-of-types.html#_schedule_for_removal_of_mapping_types](https://www.elastic.co/guide/en/elasticsearch/reference/6.x/removal-of-types.html#_schedule_for_removal_of_mapping_types))
All hopes on Mongo team

**Comment by snoopy83101** [comment]
No, I tried elastic5 doc manager not work。
error': {u'reason': u'Rejecting mapping update to [geek001] as the final mapping would have more than 1 type: [test, tokens]'
OplogThread: Failed during dump collection cannot recover!

**Comment by andrei1610399** [comment]
Elastic5 doc manager works fine for me. I have upgraded ES to 6.2.2 two days ago, started mongo-connector with elastic5 doc manager and don't see any problems. I store one document type per index though. In my scenario I create index from database name + collection name, like "my_db_my_coolection". This is done specifically to have one type of document in one index.

**Comment by atwebm** [comment]
@andrei1610399 How have you configured it? I'm very interested to use this solution for our own collections.

**Comment by andrei1610399** [comment]
The configuration didn't work for me. I don't know the collection name in advance. I have changed the elastic5 doc manager code a little. The doc manager receives the namespace as a parameter for it's functions. The namespace (ns)  looks like db_name.collection_name. I replace it with db_name_collection_name.collection_name. As a result of this change the index is created with name db_name_collection_name and the document name remains collection_name.

**Comment by hackrflov** [comment]
I got the same matter, solved by downgrading es to 5.6.3.
